examples/vectoradd_torch/tesseract_api.py

```diff
@@ -0,0 +1,222 @@
+    def validate_shape_inputs(self) -> Self:
+        if self.a.v.shape != self.b.v.shape:
+            raise ValueError(
+                f"a.v and b.v must have the same shape. "
+                f"Got {self.a.v.shape} and {self.b.v.shape} instead."
+            )
+        return self
+
+
+class Result_and_Norm(BaseModel):
+    result: Differentiable[Array[(None,), Float32]] = Field(
+        description="Vector s_a·a + s_b·b"
+    )
+    normed_result: Differentiable[Array[(None,), Float32]] = Field(
+        description="Normalized Vector s_a·a + s_b·b/|s_a·a + s_b·b|"
+    )
+
+
+class OutputSchema(BaseModel):
+    vector_add: Result_and_Norm
+    vector_min: Result_and_Norm
+
+
+#
+# Required endpoints
+#
+
+
+def evaluate(inputs: Any) -> Any:
+    a_scaled = inputs["a"]["s"] * inputs["a"]["v"]
+    b_scaled = inputs["b"]["s"] * inputs["b"]["v"]
+    add_result = a_scaled + b_scaled
+    min_result = a_scaled - b_scaled
+    return {
+        "vector_add": {
+            "result": add_result,
+            "normed_result": add_result / torch.linalg.norm(add_result, ord=2),
+        },
+        "vector_min": {
+            "result": min_result,
+            "normed_result": min_result / torch.linalg.norm(min_result, ord=2),
+        },
+    }
+
+
+def apply(inputs: InputSchema) -> OutputSchema:
+    # Optional: Insert any pre-processing/setup that doesn't require tracing
+    # and is only required when specifically running your apply function
+    # and not your differentiable endpoints.
+    # For example, you might want to set up a logger or mlflow server.
+    # Pre-processing should not modify any input that could impact the
+    # differentiable outputs in a nonlinear way (a constant shift
+    # should be safe)
+
+    # Convert to pytorch tensors to enable torch.jit
+    tensor_inputs = tree_map(convert_to_tensors, inputs.model_dump())
+    out = evaluate(tensor_inputs)
+
+    # Optional: Insert any post-processing that doesn't require tracing
+    # For example, you might want to save to disk or modify a non-differentiable
+    # output. Again, do not modify any differentiable output in a non-linear way.
+    return out
+
+
+#
+# Pytorch-handled AD endpoints (no need to modify)
+#
+
+
+def jacobian(
+    inputs: InputSchema,
+    jac_inputs: set[str],
+    jac_outputs: set[str],
+):
+    # convert all numbers and arrays to torch tensors
+    tensor_inputs = tree_map(convert_to_tensors, inputs.model_dump())
+
+    # flatten the dictionaries such that they can be accessed by paths
+    path_inputs = flatten_with_paths(tensor_inputs, jac_inputs)
+
+    # transform the dictionaries into a list of values for a positional function
+    pos_inputs = path_inputs.values()
+    keys = path_inputs.keys()
+
+    # create a positional function that accepts a list of values and returns a set of tuples
+    filtered_pos_eval = filter_pos_func(
+        evaluate, tensor_inputs, jac_outputs, keys, output_to_tuple=True
+    )
```

Comment:
We already know what the keys will be (similar elsewhere)
```suggestion
    # flatten the dictionaries such that they can be accessed by paths
    pos_inputs = flatten_with_paths(tensor_inputs, jac_inputs).values()

    # create a positional function that accepts a list of values and returns a set of tuples
    filtered_pos_eval = filter_pos_func(
        evaluate, tensor_inputs, jac_outputs, jac_inputs, output_to_tuple=True
    )
```

Reply:
changed ! :)